File: source/kCura.Relativity.Export/Service/CaseManager.vb

Imports Relativity

Namespace kCura.Relativity.Export.Service
	Public Class CaseManager
		Inherits kCura.EDDS.WebAPI.CaseManagerBase.CaseManager

		Public Sub New(ByVal credentials As Net.ICredentials, ByVal cookieContainer As System.Net.CookieContainer)
			MyBase.New()

			Me.Credentials = credentials
			Me.CookieContainer = cookieContainer
			Me.Url = String.Format("{0}CaseManager.asmx", Config.WebServiceURL)
			Me.Timeout = Settings.DefaultTimeOut
		End Sub

		Protected Overrides Function GetWebRequest(ByVal uri As System.Uri) As System.Net.WebRequest
			Dim wr As System.Net.HttpWebRequest = DirectCast(MyBase.GetWebRequest(uri), System.Net.HttpWebRequest)
			wr.UnsafeAuthenticatedConnectionSharing = True
			wr.Credentials = Me.Credentials
			Return wr
		End Function

		Public Shared Function ConvertToCaseInfo(ByVal toConvert As kCura.EDDS.WebAPI.CaseManagerBase.CaseInfo) As CaseInfo
			Dim c As New CaseInfo
			With toConvert
				c.ArtifactID = .ArtifactID
				c.MatterArtifactID = .MatterArtifactID
				c.Name = .Name
				c.RootArtifactID = .RootArtifactID
				c.RootFolderID = .RootFolderID
				c.StatusCodeArtifactID = .StatusCodeArtifactID
				c.EnableDataGrid = .EnableDataGrid
				c.DocumentPath = .DocumentPath
				c.DownloadHandlerURL = .DownloadHandlerURL
			End With
			Return c
		End Function

#Region " Shadow Functions "
		Public Shadows Function RetrieveAll() As System.Data.DataSet
			Dim tries As Int32 = 0
			While tries < Config.MaxReloginTries
				tries += 1
				Try
					If kCura.WinEDDS.Config.UsesWebAPI Then
						Return MyBase.RetrieveAllEnabled()
					Else
						'Return _caseManager.RetrieveAll(_identity).ToDataSet()
					End If
				Catch ex As System.Exception
					If TypeOf ex Is System.Web.Services.Protocols.SoapException AndAlso ex.ToString.IndexOf("NeedToReLoginException") <> -1 AndAlso tries < Config.MaxReloginTries Then
						Helper.AttemptReLogin(Me.Credentials, Me.CookieContainer, tries, False)
					Else
						Throw
					End If
				End Try
			End While
			Return Nothing
		End Function

		Public Shadows Function Read(ByVal caseArtifactID As Int32) As Relativity.CaseInfo
			Dim tries As Int32 = 0
			While tries < Config.MaxReloginTries
				tries += 1
				Try
					Return ConvertToCaseInfo(MyBase.Read(caseArtifactID))
				Catch ex As System.Exception
					If TypeOf ex Is System.Web.Services.Protocols.SoapException AndAlso ex.ToString.IndexOf("NeedToReLoginException") <> -1 Then
						If tries < Config.MaxReloginTries Then
							Helper.AttemptReLogin(Me.Credentials, Me.CookieContainer, tries, False)
						Else
							Throw ex
						End If
					Else
						Throw
					End If
				End Try
			End While
			Return Nothing
		End Function
#End Region

	End Class
End Namespace

File: source/kCura.Relativity.Export.ConvertedToC#/Service/CaseManager.cs

using System;

using CaseInfo = Relativity.CaseInfo;

namespace kCura.Relativity.Export.Service
{
	public class CaseManager : kCura.EDDS.WebAPI.CaseManagerBase.CaseManager
	{

		public CaseManager(System.Net.ICredentials credentials, System.Net.CookieContainer cookieContainer) : base()
		{

			this.Credentials = credentials;
			this.CookieContainer = cookieContainer;
			this.Url = string.Format("{0}CaseManager.asmx", Config.WebServiceURL);
			this.Timeout = Settings.DefaultTimeOut;
		}

		protected override System.Net.WebRequest GetWebRequest(System.Uri uri)
		{
			System.Net.HttpWebRequest wr = (System.Net.HttpWebRequest)base.GetWebRequest(uri);
			wr.UnsafeAuthenticatedConnectionSharing = true;
			wr.Credentials = this.Credentials;
			return wr;
		}

		public static CaseInfo ConvertToCaseInfo(kCura.EDDS.WebAPI.CaseManagerBase.CaseInfo toConvert)
		{
			CaseInfo c = new CaseInfo();
			var _with1 = toConvert;
			c.ArtifactID = _with1.ArtifactID;
			c.MatterArtifactID = _with1.MatterArtifactID;
			c.Name = _with1.Name;
			c.RootArtifactID = _with1.RootArtifactID;
			c.RootFolderID = _with1.RootFolderID;
			c.StatusCodeArtifactID = _with1.StatusCodeArtifactID;
			c.EnableDataGrid = _with1.EnableDataGrid;
			c.DocumentPath = _with1.DocumentPath;
			c.DownloadHandlerURL = _with1.DownloadHandlerURL;
			return c;
		}

		#region " Shadow Functions "
		public new System.Data.DataSet RetrieveAll()
		{
			Int32 tries = 0;
			while (tries < Config.MaxReloginTries) {
				tries += 1;
				try {
					if (Config.UsesWebAPI) {
						return base.RetrieveAllEnabled();
					} else {
						//Return _caseManager.RetrieveAll(_identity).ToDataSet()
					}
				} catch (System.Exception ex) {
					if (ex is System.Web.Services.Protocols.SoapException && ex.ToString().IndexOf("NeedToReLoginException") != -1 && tries < Config.MaxReloginTries) {
						//Helper.AttemptReLogin(this.Credentials, this.CookieContainer, tries, false);
					} else {
						throw;
					}
				}
			}
			return null;
		}

		public new CaseInfo Read(Int32 caseArtifactID)
		{
			Int32 tries = 0;
			while (tries < Config.MaxReloginTries) {
				tries += 1;
				try {
					return ConvertToCaseInfo(base.Read(caseArtifactID));
				} catch (System.Exception ex) {
					if (ex is System.Web.Services.Protocols.SoapException && ex.ToString().IndexOf("NeedToReLoginException") != -1) {
						if (tries < Config.MaxReloginTries) {
							//Helper.AttemptReLogin(this.Credentials, this.CookieContainer, tries, false);
						} else {
							throw ex;
						}
					} else {
						throw;
					}
				}
			}
			return null;
		}
		#endregion

	}
}
